src/main/java/GUI/ButtonFactory.java

```diff
@@ -0,0 +1,103 @@
+package GUI;
+
+import ButtonMappings.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import javax.swing.*;
+
+/**
+ * ButtonFactory is a class that creates all the button instances that need to be
+ * displayed to the screen with their respective action handling
+ */
+public class ButtonFactory {
+    /**
+     * InstanceVar buttonStrings: an ArrayList of Strings that contains all the output
+     * strings that need to be displayed on each button
+     * InstanceVar actionPerformed: A Mapping between the button string and the number
+     * of times the button was pressed
+     * InstanceVar buttonMappings: an ArrayList of ButtonHandlers, which contains all the
+     * different Button'x' objects to determine user input
+     */
+    private ArrayList<String> buttonStrings;
+    private HashMap<String, Integer> actionPerformed;
+    private ArrayList<ButtonMapping> buttonMappings;
+
+    /**
+     * Constructor with no parameters for this class
+     */
+    public ButtonFactory (){
+        this.buttonStrings = new ArrayList<>();
+        this.actionPerformed = new HashMap<>();
+        this.buttonMappings = new ArrayList<>();
+        addButtons();
+    }
+    public void addStrings(ArrayList<String> options){
+        this.buttonStrings = options;
+    }
+
+    /**
+     * Function that creates all the ButtonHandlers and adds them to the Arraylist
+     * of ButtonHandlers
+     */
+    public void addButtons(){
+        Button1 b1 = new Button1();
```

Comment:
Maybe we could refractor this by extracting a method or using a four loop to create and add 16 buttons